test/statetest/statetest.hpp

```diff
@@ -59,8 +63,59 @@ struct StateTransitionTest
     std::unordered_map<uint64_t, std::string> input_labels;
 };
 
+template <typename T>
+T from_json(const json::json& j) = delete;
+
+template <>
+state::BlockInfo from_json<state::BlockInfo>(const json::json& j);
+
+template <>
+state::State from_json<state::State>(const json::json& j);
+
+template <>
+state::Transaction from_json<state::Transaction>(const json::json& j);
+
 StateTransitionTest load_state_test(const fs::path& test_file);
 
 void run_state_test(const StateTransitionTest& test, evmc::VM& vm);
 
+namespace utils
+{
+inline size_t countl_zero_bytes(uint16_t x)
+{
+    auto upper = uint8_t(x >> 8);
+    auto lower = uint8_t(x & 0xFF);
+    return upper ? 0 : (lower ? 1 : 2);
+}
+
+inline size_t countl_zero_bytes(uint32_t x)
+{
+    auto upper = uint16_t(x >> 16);
+    auto lower = uint16_t(x & 0xFFFF);
+    return upper ? countl_zero_bytes(upper) : 2 + countl_zero_bytes(lower);
+}
+
+inline size_t countl_zero_bytes(uint64_t x)
+{
+    auto upper = uint32_t(x >> 32);
+    auto lower = uint32_t(x & 0xFFFFFFFF);
+    return upper ? countl_zero_bytes(upper) : 4 + countl_zero_bytes(lower);
+}
+
+inline std::string hex(uint64_t v)
```

Comment:
I'd call this something like `shortest_hex` or something similar to signal intent.